tuta-sdk/rust/src/crypto/crypto_facade.rs

```diff
@@ -0,0 +1,231 @@
+#![allow(dead_code)] // TODO: remove this later
+
+use std::collections::HashMap;
+use std::sync::Arc;
+use zeroize::Zeroizing;
+use crate::crypto::aes::{aes_128_decrypt_no_padding_fixed_iv, aes_256_decrypt_no_padding};
+use crate::crypto::ecc::EccPublicKey;
+use crate::crypto::rsa::RSAEncryptionError;
+use crate::crypto::tuta_crypt::{PQError, PQMessage};
+use crate::element_value::{ElementValue, GeneratedId, ParsedEntity};
+use crate::metamodel::TypeModel;
+use super::key_loader_facade::{AsymmetricKeyPair, GenericAesKey, GenericAesKeyData, KeyLoaderFacade, KeyLoadError};
+
+const OWNER_ENC_SESSION_KEY_NAME: &'static str = "_ownerEncSessionKey";
+const OWNER_KEY_VERSION_NAME: &'static str = "_ownerKeyVersion";
+const OWNER_GROUP_NAME: &'static str = "_ownerGroup";
+
+macro_rules! get_field {
```

Comment:
it can be just a generic function and then you might not need to specify the type most of the time